op/pulse.c

```diff
@@ -175,6 +179,20 @@ static void _pa_sink_input_info_cb(pa_context *c,
 	}
 }
 
+static void _pa_server_info_cb(pa_context *c,
+			       const pa_server_info *i,
+			       void *data)
+{
+	is_pipewire = false;
+	if (i) {
+		if (strstr(i->server_name, "PipeWire") != NULL) {
+			// server is PipeWire
+			d_print("Pulseaudio server is pipewire. Disabling _pa_stream_drain()\n");
```

Comment:
Is there any upstream issue we can reference here, so that we know when we can drop the workaround?

Reply:
Hi. I really can't investigate this issue right now. I live in Iran and ... nothing good is going on right now. A few weeks ago the police was in our university (Sharif University of Technology. Go look at its wikipedia page) shooting students and beating faculty members (yeah our own professors) and arresting people and ... I just need some time to deal with this shit.

Reply:
take care :(

Reply:
Hope all is well.

I don't think there is an existing upstream report other than the original [pipewire/-/issues/946](https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/946), but the change that broke it again ([pipewire/-/commit/b9c8f0f6b9](https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/b9c8f0f6b9dd479a0372adae514653509ff1d957)) says it's expected behavior:

> A corked stream should not complete a pending drain operation

I experimented with uncorking the stream first and it also solves the issue, but just removing the call makes more sense to me. It doesn't seem like we ever had any reason to wait for the stream to drain in the first place.

Reply:
So, let's go with #1268?